backend/assignment-1-cezaram28/Assignment 1/src/main/java/com/cezaram28/Assignment1/controller/AnswersController.java

```diff
@@ -0,0 +1,41 @@
+package com.cezaram28.Assignment1.controller;
+
+import com.cezaram28.Assignment1.dto.AnswerDTO;
+import com.cezaram28.Assignment1.dto.UserDTO;
+import com.cezaram28.Assignment1.service.AnswerManagementService;
+import com.cezaram28.Assignment1.service.QuestionManagementService;
+import com.cezaram28.Assignment1.service.UserManagementService;
+import lombok.RequiredArgsConstructor;
+import org.springframework.web.bind.annotation.*;
+
+import java.util.List;
+
+@RestController
+@RequiredArgsConstructor
+public class AnswersController {
+    private final AnswerManagementService answerService;
+    private final QuestionManagementService questionManagementService;
+    private final UserManagementService userManagementService;
+
+    @GetMapping("questions/{id}/answers")
+    public List<AnswerDTO> getByQuestion(@PathVariable int id) {
+        List<AnswerDTO> dto = answerService.listAllToQuestion(id);
+        return dto;
+    }
+
+    @PostMapping("/answers")
+    public AnswerDTO addAnswer(@RequestBody AnswerDTO answerDTO) {
+        return answerService.addAnswer(answerDTO);
+    }
+
+    @DeleteMapping("/answers/{id}")
+    public void delete(@PathVariable int id, @RequestBody UserDTO userDTO) {
+        answerService.removeAnswer(id, userDTO);
+    }
+
+    @PutMapping("/answers/{userId}")
```

Comment:
Instead of the userId, you should pass the answer ID (https://slides.com/spet/utcn-sd-assignment-3#/3/2), otherwise it can be very misleading. You could get the user name from the spring security context: https://github.com/serban-petrescu/utcn-sd-resources/blob/master/examples/assignment-3/backend/src/main/java/ro/utcn/sd/a3/service/TeacherUserDetailsService.java#L33